static/app/views/settings/organizationIntegrations/repositoryProjectPathConfigForm.tsx

```diff
@@ -38,33 +39,37 @@ function RepositoryProjectPathConfigForm({
   projects,
   repos,
 }: Props) {
-  const api = useApi();
   const formRef = useRef(new FormModel());
   const repoChoices = repos.map(({name, id}) => ({value: id, label: name}));
+  const [selectedRepo, setSelectedRepo] = useState<GeneralSelectValue | null>(null);
 
   /**
-   * Automatically switch to the default branch for the repo
+   * Using the integration repo search, automatically switch to the default branch for the repo,
+   * once one is selected in the form.
    */
-  function handleRepoChange(id: string) {
-    const repo = repos.find(r => r.id === id);
-    if (!repo) {
-      return;
+  const {data: integrationReposData} = useApiQuery<{repos: IntegrationRepository[]}>(
```

Comment:
i did take a look at some examples around the codebase of `useApiQuery` usage, but please give me a sanity check that this is in-line with our conventions

Reply:
seems fine